sasautos/h54s.sas

```diff
@@ -475,85 +475,50 @@ options NOQUOTELENMAX LRECL=32000 spool;
     %return;
   %end;
 
-
-  data _null_;
-    call symput('qc', '"');
-    call symput('pf', "%upcase(&dsn)");
-    call symput('dc', '$');
-    call symput('dt_', 'dt_');
+  proc contents noprint data=&libn..&dsn 
+    out=tempCols(keep=name type length varnum);
   run;
-
-  proc sql;
-    create table tempCols as
-    select upcase(name) as name, type, length from dictionary.columns 
-    where upcase(memname)="%upcase(&dsn)" and libname="%upcase(&libn)";
-  quit;
-
-  %let totalCols = &sqlObs;
-
-  proc sql;
-    select trim(name), trim(type), length into :name1-:name999, :type1-:type999, :length1-:length999
-    from tempCols;
-  quit;
+  
+  proc sort data=tempCols; by varnum; run;
```

Comment:
adding proc sort back in - this retains the variable order, with minimal overhead.  I know variable order shouldn't matter, but sometimes.. it just does!